sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpAnnotatedMessageExtensions.cs

```diff
@@ -200,6 +207,27 @@ public static BinaryData GetEventBody(this AmqpAnnotatedMessage instance)
             return defaultValue;
         }
 
+        /// <summary>
+        ///   Retrieves the replication segment of an event from an <see cref="AmqpAnnotatedMessage" />.
+        /// </summary>
+        ///
+        /// <param name="instance">The instance that this method was invoked on.</param>
+        /// <param name="defaultValue">The value to return if the replication segment is not represented in the <paramref name="instance"/>.</param>
+        ///
+        /// <returns>The replication segment, if represented in the <paramref name="instance"/>; otherwise, <paramref name="defaultValue"/>.</returns>
+        ///
+        public static string GetReplicationSegment(this AmqpAnnotatedMessage instance,
+                                     string defaultValue = "-1")
+        {
+            if ((instance.HasSection(AmqpMessageSection.MessageAnnotations))
+                && (instance.MessageAnnotations.TryGetValue(AmqpProperty.ReplicationSegment.ToString(), out var value)))
+            {
+                return ((int)value).ToString();
```

Comment:
TODO : the value being returned is an int and I think it should be a stream. This line currently does not work with mocking so I need to either figure out why value is an int or make some kind of if statement? I didn't think the service doc was super clear about what type the replication segment should be.